packages/core/testing/src/r3_test_bed.ts

```diff
@@ -402,17 +433,67 @@ export class TestBedRender3 implements TestBed {
   }
 
   private destroyActiveFixtures(): void {
+    let errorCount = 0;
     this._activeFixtures.forEach((fixture) => {
       try {
         fixture.destroy();
       } catch (e) {
+        errorCount++;
         console.error('Error during cleanup of component', {
           component: fixture.componentInstance,
           stacktrace: e,
         });
       }
     });
     this._activeFixtures = [];
+
+    if (errorCount > 0 && this.shouldRethrowTeardownErrors()) {
+      throw Error(
+          `${errorCount} ${(errorCount === 1 ? 'component' : 'components')} ` +
+          `threw errors during cleanup`);
+    }
+  }
+
+  private shouldRethrowTeardownErrors() {
+    const instanceOptions = this._instanceTeardownOptions;
+    const environmentOptions = TestBedRender3._environmentTeardownOptions;
+
+    // If the new teardown behavior hasn't been configured, preserve the old behavior.
+    if (!instanceOptions && !environmentOptions) {
+      return false;
+    }
+
+    // Otherwise use the configured behavior or default to rethrowing.
+    return instanceOptions?.rethrowErrors ?? environmentOptions?.rethrowErrors ?? true;
+  }
+
+  shouldTearDownTestingModule(): boolean {
+    return this._instanceTeardownOptions?.destroyAfterEach ??
+        TestBedRender3._environmentTeardownOptions?.destroyAfterEach ?? false;
+  }
+
+  tearDownTestingModule() {
+    // If the module ref has already been destroyed, we won't be able to get a test renderer.
+    if (this._testModuleRef === null) {
+      return;
+    }
+    // Resolve the renderer ahead of time, because we want to remove the root elements as the very
+    // last step, but the injector will be destroyed as a part of the module ref destruction.
+    const testRenderer = this.inject(TestComponentRenderer);
+    try {
+      this._testModuleRef.destroy();
+    } catch (e) {
+      if (this.shouldRethrowTeardownErrors()) {
+        throw e;
+      } else {
+        console.error('Error during cleanup of module', {
+          component: this._testModuleRef.instance,
+          stacktrace: e,
+        });
+      }
+    } finally {
+      testRenderer.removeAllRootElements();
```

Comment:
Given that the `TestComponentRenderer` class can be configured via DI, it may not have the `removeAllRootElements` function:
```suggestion
      testRenderer.removeAllRootElements?.();
```

Reply:
I'll mark it as optional, although this interface is private.